Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -408,6 +444,44 @@ namespace xr
             PopulateSwapchains(primaryRenderResource.ViewState);
         }
 
+        void InitializeHandResources()
+        {
+            if (!HandData.supportsHandTracking)
+            {
+                return;
+            }
+
+            constexpr std::array<XrHandEXT, 2> HANDEDNESS_EXT{XR_HAND_LEFT_EXT, XR_HAND_RIGHT_EXT};
+            XrHandTrackerCreateInfoEXT trackerCreateInfo{XR_TYPE_HAND_TRACKER_CREATE_INFO_EXT};
+            trackerCreateInfo.handJointSet = XR_HAND_JOINT_SET_DEFAULT_EXT;
+
+            for (int i = 0; i < HandData.handsInfo.size(); i++)
+            {
+                // Create the hand trackers
+                HandData.handsInfo[i].hand = HANDEDNESS_EXT[i];
+                trackerCreateInfo.hand = HANDEDNESS_EXT[i];
+                XrCheck(HmdImpl.Extensions->xrCreateHandTrackerEXT(Session, &trackerCreateInfo, &HandData.handsInfo[i].handTracker));
+                    
+                // Preallocate joint buffers
+                HandData.handsInfo[i].handJointData.InitializeXRHandJoints();
+            }
+
+            HandData.HandsInitialized = true;
+        }
+
+        void UninitializeHandResources()
+        {
+            if (HandData.HandsInitialized)
+            {
+                for (HandInfo handInfo : HandData.handsInfo)
+                {
+                    HmdImpl.Extensions->xrDestroyHandTrackerEXT(handInfo.handTracker);
```

Comment:
This looks awfully like something that could live in an object destructor and go for an RAII pattern.